src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -2426,9 +2437,9 @@ private void doApplicationData(
                 {
                     assert AmqpState.initialOpening(state);
 
-                    final DirectBuffer buffer = payload.buffer();
-                    final int offset = payload.offset();
-                    final int limit = payload.limit();
+                    final DirectBuffer buffer = payload.sizeof() > 0 ? payload.buffer() : null;
+                    final int offset = payload.sizeof() > 0 ? payload.offset() : -1;
+                    final int limit = payload.sizeof() > 0 ? payload.limit() : -1;
```

Comment:
Shouldn't this be a test for `payload != null` instead, passing `null` for `payload` to `doApplicationData` when needed?

With feedback above for `doData` applied, then `payload` would pass directly to `doData` without requiring these `null` checks.

Reply:
When we get `decodeTransfer` with a null value (`0x40`), the payload won't be null and will be empty octets in the current implementation of `AmqpMessageDecoder` (`messageFragmentRW.build()` is always called so it won't be null).

Reply:
Yes, but by the time it arrives here in `doApplicationData`, doesn't the payload represent the `DATA` frame payload, which means it should already be `null` before being passed?